src/js/app.js

```diff
@@ -29,21 +42,38 @@ class App {
     };
     this.submitHandler = {
       "purchase-input-form": () => {
-        const isValid = this.#model.purchaseLotto(
-          this.#view.$amountInput.value
+        const hasManualLottos = this.#model.state.manualLottos.length > 0;
+        const isValidManualLottoNumbers =
+          this.#model.isValidManualLottoNumbers();
+        const isValidAmount = this.#model.isValidAmount(
+          Number(this.#view.$amountInput.value)
         );
 
-        if (!isValid) return;
+        if (!isValidAmountUnit(this.#view.$amountInput.value)) {
+          alert(MESSAGE_ABOUT_UNIT_OF_AMOUNT);
+          return;
+        }
+
+        if (hasManualLottos && !isValidManualLottoNumbers) {
+          alert(MESSAGE_ABOUT_DUPLICATION_NUMBER);
+          return;
+        }
+
+        if (!isValidAmount) {
+          alert(MESSAGE_ABOUT_ENTERED_OUTSTANDING_AMOUNT);
+          return;
+        }
```

Comment:
validation 함수에서 Error를 던져주고 try catch에서 에러를 처리하는게 더 일관성이 있을 것 같아요.
```suggestion
        try {
          checkAmountUnitValid(this.#view.$amountInput.value);
          checkManualLottos();
          checkAmountValid();
        } catch (error) {
          if (error instanceof CustomError) {
             alert(error.message);
          }
          // 프로덕션이라면 에러 모니터링 시스템에 capture.. 
        }
```

Reply:
try/catch는 예외 처리 하기 쉽게하기 위해 나온 친구인데, 단순히 **비동기 처리할때만** 사용해야지라고 고정 관념에 빠져있었네요.
너무 안일했네요 ㅠㅠ 깨닫게 해주셔서 감사합니다!

리뷰 반영 커밋
https://github.com/next-step/js-lotto/pull/208/commits/d708b40ca68593d37fa16e13be2578822547542f